File: src/test/java/io/binac/leetcode/InterleavingStringTest.java
package io.binac.leetcode;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class InterleavingStringTest {
    private final InterleavingString.Solution1 solution1 = new InterleavingString.Solution1();

    @Test
    void test1() {
        assertTrue(solution1.isInterleave("aabcc", "dbbca", "aadbbcbcac"));
        assertFalse(solution1.isInterleave("aabcc", "dbbca", "aadbbbaccc"));
    }
}

File: cpp/unittests/InterleavingStringTest.cpp
#include "InterleavingString.h"
#include "gtest/gtest.h"

using namespace lcpp;

TEST(InterleavingString, Solution97_1) {
  EXPECT_TRUE(Solution97_1().isInterleave("aabcc", "dbbca", "aadbbcbcac"));
  EXPECT_FALSE(Solution97_1().isInterleave("aabcc", "dbbca", "aadbbbaccc"));
}
